db/flush_job.cc

```diff
@@ -384,7 +384,7 @@ Status FlushJob::WriteLevel0Table() {
           cfd_->GetName(), existing_snapshots_,
           earliest_write_conflict_snapshot_, snapshot_checker_,
           output_compression_, mutable_cf_options_.sample_for_compression,
-          cfd_->ioptions()->compression_opts,
+          mutable_cf_options_.compression_opts,
```

Comment:
`mutable_cf_options_` is a const reference, and `BuildTable()` does not hold the db mutex. What if a user thread calls `SetOptions` while flush job is accessing `mutable_cf_options_.compression_opts`?

Reply:
nvm, we do a copy before calling in `FlushMemTableToOutputFile` and `AtomicFlushMemTablesToOutputFiles`.